controllers/restore.go

```diff
@@ -839,3 +845,164 @@ func setUserRestoreFilters(
 		}
 	}
 }
+
+// delete all velero restores created by the acmRestore object
+// and remove finalizers
+func finalizeRestore(
+	ctx context.Context,
+	c client.Client,
+	acmRestore *v1beta1.Restore,
+	veleroRestoreList veleroapi.RestoreList,
+) error {
+
+	reqLogger := log.FromContext(ctx)
+
+	for _, veleroRestore := range veleroRestoreList.Items {
+		if err := c.Delete(ctx, &veleroRestore); err != nil {
```

Comment:
should probably check the error here to make sure it's not a NotFoundError, otherwise we may error out if the velerorestore is already gone

Reply:
It doesn't matter if we get an error, all we need here is to mark the velero restores for deletion